doc/code/qml_pauli.rst

```diff
@@ -0,0 +1,94 @@
+the :math:`i^{th}` qubit. The full :math:`n`-qubit Pauli group has size
+:math:`4^n` (neglecting the four possible global phases that may arise from
+multiplication of its elements).
+
+.. currentmodule:: pennylane.pauli
+
+.. automodapi:: pennylane.pauli.pauli
+    :no-inheritance-diagram:
+    :no-inherited-members:
+    :skip: binary_to_pauli, pauli_to_binary, are_identical_pauli_words
+
+.. automodapi:: pennylane.pauli.pauli_utils
+    :no-inheritance-diagram:
+    :no-inherited-members:
+    :skip: reduce
+
+Grouping observables
+--------------------
+
+Pauli words can be used for expressing a qubit :class:`~pennylane.Hamiltonian`.
+A qubit Hamiltonian has the form :math:`H_{q} = \sum_{J} C_J P_J` where
+:math:`C_{J}` are numerical coefficients, and :math:`P_J` are Pauli words.
+
+A list of Pauli words can be partitioned according to certain grouping
+strategies. As an example, the :func:`~.group_observables` function partitions
+a list of observables (Pauli operations and tensor products thereof) into
+groupings according to a binary relation (e.g., qubit-wise commuting):
+
+>>> observables = [qml.PauliY(0), qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(1)]
+>>> obs_groupings = group_observables(observables)
+>>> obs_groupings
+[[Tensor(PauliX(wires=[0]), PauliX(wires=[1]))],
+ [PauliY(wires=[0]), PauliZ(wires=[1])]]
+
+The :math:`C_{J}` coefficients for each :math:`P_J` Pauli word making up a
+Hamiltonian can also be specified along with further options, such as the
+Pauli-word grouping method (e.g., *qubit-wise commuting*) and the underlying
+graph-colouring algorithm (e.g., *recursive largest first*) used for creating
+the groups of observables:
+
+>>> obs = [qml.PauliY(0), qml.PauliX(0) @ qml.PauliX(1), qml.PauliZ(1)]
+>>> coeffs = [1.43, 4.21, 0.97]
+>>> obs_groupings, coeffs_groupings = group_observables(obs, coeffs, 'qwc', 'rlf')
+>>> obs_groupings
+[[Tensor(PauliX(wires=[0]), PauliX(wires=[1]))],
+ [PauliY(wires=[0]), PauliZ(wires=[1])]]
+>>> coeffs_groupings
+[[4.21], [1.43, 0.97]]
+
+.. currentmodule:: pennylane.pauli
+
+.. automodapi:: pennylane.pauli.graph_colouring
+    :no-inheritance-diagram:
+    :no-inherited-members:
+
+.. automodapi:: pennylane.pauli.group_observables
+    :no-inheritance-diagram:
+    :no-inherited-members:
+    :skip: Wires, observables_to_binary_matrix, qwc_complement_adj_matrix, binary_to_pauli
+    :skip: are_identical_pauli_words, largest_first, recursive_largest_first
+
+.. automodapi:: pennylane.pauli.grouping_utils
+    :no-inheritance-diagram:
+    :no-inherited-members:
+    :skip: pauli_to_binary
+
+.. automodapi:: pennylane.pauli.optimize_measurements
+    :no-inheritance-diagram:
+    :no-inherited-members:
+    :skip: diagonalize_qwc_groupings, group_observables
+
+.. automodapi:: pennylane.pauli.transformations
+    :no-inheritance-diagram:
+    :no-inherited-members:
+    :skip: are_identical_pauli_words, is_pauli_word, is_qwc, pauli_to_binary, template
```

Comment:
I had a look in `pauli/__init__.py`, and it looks like the functions in `optimize_measurements.py`, `group_observables.py`, `transformations.py`, `pauli_utils.py`, `pauli.py` etc., are all imported top level into the `qml.pauli` namespace?

If so, you could just generate a single autosummary table by deleting the individual calls above, and simply having

```rest
.. automodapi:: pennylane.pauli
    :no-inheritance-diagram:
    :no-inherited-members:
```

The only one where this is not the case is `graph_colouring`, so that one will need to remain